packages/Filter/src/helpers/filterData.js

```diff
@@ -1,18 +1,29 @@
+import Filter from "../Filter";
 import testers from "./testers";
 import { logicalFilterOperators } from "../rules";
 
 export default function filterData({ filters, operator, columns, data }) {
   if (filters.length === 0) return data;
 
   return data.filter(row => {
-    const tester = filter =>
-      filter.rule
-        ? testers[filter.rule](
-            row[filter.columnId],
-            filter.value,
-            columns.find(column => filter.columnId === column.id)
-          )
-        : true;
+    function tester(filter) {
+      if (filter.rule) {
+        const columnType = columns.find(column => column.id === filter.columnId).type;
+        switch (columnType) {
+          case Filter.types.columnTypes.SINGLE_SELECT:
+          case Filter.types.columnTypes.MULTI_SELECT:
+            return testers[filter.rule](row[filter.columnId].id, filter.value);
```

Comment:
the tester function --`testers[filter.rule]` doesn't need the 3rd parameter (`column`)?

Reply:
Not for the rules used by the SINGLE and MULTI selects